packages/wrangler/src/api/startDevWorker/DevEnv.ts

```diff
@@ -93,6 +95,34 @@ export class DevEnv extends EventEmitter {
 		});
 	}
 
+	/**
+	 * Awaits for all the local runtime controllers to be ready, and when they are
+	 * it resolves a promise on the proxy controller to let it know that the
+	 * local server is now ready to handle requests
+	 */
+	private async resolveAllRuntimeControllersReady(): Promise<void> {
```

Comment:
Why is this needed? the proxy controller already receives `reloadComplete` events from the runtime controllers: https://github.com/cloudflare/workers-sdk/pull/9910/files#diff-49321de92dbc2c86de8aea8dc98fb93bc26a4ff1a2e491844f5f6b6ad0bb6b55R87

Reply:
What about resolving the proxy controller promise based on the first reload complete event received in the proxy controller? Then you don't need the new teardown event or any of this logic, I don't think

Reply:
> What about resolving the proxy controller promise based on the first reload complete event received in the proxy controller? Then you don't need the new teardown event or any of this logic, I don't think

`DevEnv` has in theory an arbitrary number of runtime controllers, do you think that it is always guaranteed that only one will work at a time? If that's the case then yeah we could just listen for the first `reloadComplete` event I'd reckon. Otherwise from a generic perspective checking that all the runtime controllers are done seems the more correct approach to me.

Reply:
ok, sounds good, thanks for the clarification, I'll update the code 🙂👍